opm/parser/eclipse/EclipseState/IOConfig/IOConfig.cpp

```diff
@@ -109,8 +114,8 @@ namespace Opm {
                     write_restart_file = true;
                 } else {
                     std::vector<size_t>::const_iterator ci_start = timesteps.begin();
-                    int index = std::distance( ci_start, ci_timestep );
-                    if( (index % frequency) == 0) {
+                    int dist = std::distance( ci_start, ci_timestep ) + 1;
```

Comment:
For instance take the list of timesteps: " 3 4 5 6 7 8 9 12 16 19 21 27 ", frequency is 6, so every 6th of these should be written. 
The first one to be written is timestep 8, next one is 27. std::diff on iterators for timestep 8 and begin of timestelist is 5, so to get every 6th element I added the 1 to the dist integer. Please feel free to suggest a more elegant solution.